test/input_gen/recorder.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: Apache-2.0
+##
+# Copyright (C) 2020 Jihoon Lee <jhoon.it.lee@samsung.com>
+#
+# @file recorder.py
+# @brief Generate tc from given keras model
+# @author Jihoon lee <jhoon.it.lee@samsung.com>
+
+from functools import wraps
+import sys
+import os
+import warnings
+import random
+
+with warnings.catch_warnings():
+    warnings.filterwarnings("ignore", category=FutureWarning)
+    import numpy as np
+    import tensorflow as tf
+    from tensorflow.python import keras as K
+
+tf.compat.v1.enable_eager_execution()
+# Fix the seeds across frameworks
+SEED = 1234
+random.seed(SEED)
+tf.compat.v1.set_random_seed(SEED)
+np.random.seed(SEED)
+
+
+##
+# Keras Recorder
+
+##
+# @brief Record Keras model with some watchers attached
+# @note  The class might need to go through some rework for non-sequential model
+# in case of the order of graph traversal is diffrent from NNTrainer
+class KerasRecorder:
+    def __init__(
+        self,
+        file_name,
+        inputs,
+        outputs,
+        input_shape,
```

Comment:
I think you will need a batch size separately in this

Reply:
Could you explain more about this? I am not getting the benefit from separating the batchsize

Reply:
In tf.keras, input layer takes input_shape and batch_size separately.
As some tests will want to run over different batches, batch_size might be needed.

Reply:
Sounds like it is better to restructure the class to pass input_shape, label_shape and filename to `self.run` phase. To make the behavior consistent, I'll add a test using this python script first and make another PR to restructure this file to make sure things don't break down.